edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Scraper.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    public class Scraper : IMetricsScraper, IDisposable
+    {
+        const string UrlPattern = @"[^/:]+://(?<host>[^/:]+)(:[^:]+)?$";
+        static readonly Regex UrlRegex = new Regex(UrlPattern, RegexOptions.Compiled);
+        readonly HttpClient httpClient;
+        readonly Lazy<IDictionary<string, string>> endpoints;
+        readonly ILogger logger;
+
+        public Scraper(IList<string> endpoints, ILogger logger)
+        {
+            this.httpClient = new HttpClient();
+            this.endpoints = new Lazy<IDictionary<string, string>>(() => endpoints.ToDictionary(e => e, this.GetUriWithIpAddress));
+            this.logger = logger;
+        }
+
+        string GetUriWithIpAddress(string endpoint)
```

Comment:
did you format it already? I expect internal method should be put after public methods.

Reply:
stylecop isn't complaining, but I'll go ahead and move it.